File: python/palindromo.py
def es_palindromo(texto):
    texto = ''.join(c for c in texto.lower() if c.isalnum())
    return texto == texto[::-1]

palabra = input("Ingresa una palabra: ")
print("Es palíndromo" if es_palindromo(palabra) else "No es palíndromo")

File: java/palindromo.java
import java.util.Scanner;

public class palindromo {
    public static boolean esPalindromo(String texto) {
        texto = texto.toLowerCase().replaceAll("[^a-z0-9]", "");
        return texto.equals(new StringBuilder(texto).reverse().toString());
    }

    public static void main(String[] args) {
        try (Scanner scanner = new Scanner(System.in)) {
            System.out.print("Ingresa una palabra: ");
            String palabra = scanner.nextLine();
            System.out.println(esPalindromo(palabra) ? "Es palíndromo" : "No es palíndromo");
        }
    }
}